### ISSUE TITLE
Pause Media On App Pause does not seem to work on Oculus Quest 2 sleep mode

### ISSUE THREAD

**Issue by michelepanegrossi** [RenderHeads/UnityPlugin-AVProVideo#885]
**Describe the issue**
A clear and concise description of what the issue is.

MediaPlayer.PauseMediaOnAppPause = false does not seem to work on Oculus Quest 2 when the headset goes into sleep mode. I am working with an HLS video / audio stream. Is there a way to get this to work?

**Your Setup (please complete the following information):**
 - Unity version: 2020.3.9f1
 - AVPro Video version: AvPro core edition 2.1.6
 - Operating system version: Oculus Quest 2 (Android)
 - Device model: Oculus Quest 2 (Android)
 - Video specs (resolution, frame-rate, codec, file size): HLS stream, secure (HTTPS)

**To Reproduce**
1. Use the Demo 360 scene included in the package. Load a secure HLS stream in a media reference.
2. Uncheck Pause Media On App Pause in the media player options
3. Mount and unmount headset to test media playback in sleep mode

**Logs**
If applicable, add error logs to help explain your problem.

**Screenshots**

**Videos**
If applicable, add a copy of your video or the URL

### Please DO NOT LINK / ATTACH YOUR PROJECT FILES HERE
Instead email the link to us unitysupport@renderheads.com

**Comment by kahnivore** [comment]
Hi there, just to clarify - what happens when it goes into sleep mode? What do you expect to happen?

**Comment by michelepanegrossi** [comment]
Hello, thanks for replying! When the headset goes into sleep mode the streaming stops. What I would like it to do is for the stream to keep going so that when the headset is put back on the user is not behind in watching the live stream.

Alterbatively, is PauseMediaOnAppPause = false is not the way to go, is there a way of reloading the source from script to force the streaming to catch up?

Many thanks!

**Comment by michelepanegrossi** [comment]
I think the way to go would be to reload the stream every time the headset is mounted.

I am experimenting with different ways of doing this.

At the moment I am calling MediaPlayer.OpenMedia(reference) when the headset triggers a 'headset mounted' event.

However I am running into an issue where if the headset is left inactive for a longer time, say more than 15 minutes, when the headset is mounted and wakes up the live stream doesn't reload.

Should I use OnApplicationFocus instead?

What is the best way of reloading an HLS live stream so that it syncs with the current live end of the stream?

**Comment by kahnivore** [comment]
Not sure you'd need to reload the stream (if the application is still running). If you do reload it though, it should automatically start at the current 'end'.

Perhaps try using our built-in method in MediaPlayer.cs -> SeekToLiveTime, with an offset of 0. This should set the stream to be up to date. 

As for which method/even method to use - would have to investigate this. After 15 minutes when you say the live stream doesn't reload - is your headset mounted event firing or not?

**Comment by michelepanegrossi** [comment]
I am sorry for replying so late, I had to find another solution and then I missed your answer.

I needed to reaload the stream when the headset was mounted again after being unmounted and going into sleep mode. I wanted the stream to be at its live point and not where it was left when the headset went to sleep.

it would have been great to check the SeekToLiveTime option, I will go back to the project and check it when I have a moment.

I ended up forcing a reload of the stream via script, after checking the headset has gained network connectivity. I think one issue I was hacing was that AvPro was sometimes trying to load the stream before the headset had reconnected to the internet when waking from sleep.

Yes both mounted and unmounted events were being cought by the headset. I ended up using the Application Pause event to stop the playback though, as I didn't want the stream to stop when the headset was unmounted momentarily.

Thank you for your help!

Michele
